File: services/reservation/src/main/java/com/example/iter/reservation/dto/response/RentalRejectResponse.java
package com.example.iter.reservation.dto.response;

import com.example.iter.payment.api.PaymentStatus;
import com.example.iter.reservation.domain.entity.Rental;
import com.example.iter.reservation.api.RentalStatus;

public record RentalRejectResponse(
        Long rentalId,
        RentalStatus status,
        String reason,
        PaymentStatus paymentStatus
) {
    public static RentalRejectResponse of(Rental rental, PaymentStatus paymentStatus) {
        return new RentalRejectResponse(rental.getId(), rental.getStatus(), rental.getRejectReason(), paymentStatus);
    }
}

File: services/reservation/src/main/kotlin/com/example/iter/reservation/dto/response/RentalRejectResponse.kt
package com.example.iter.reservation.dto.response

import com.example.iter.payment.api.PaymentStatus
import com.example.iter.reservation.api.RentalStatus
import com.example.iter.reservation.domain.entity.Rental

data class RentalRejectResponse(
    val rentalId: Long?,
    val status: RentalStatus?,
    val reason: String?,
    val paymentStatus: PaymentStatus?,
) {
    fun rentalId(): Long? = rentalId
    fun status(): RentalStatus? = status
    fun reason(): String? = reason
    fun paymentStatus(): PaymentStatus? = paymentStatus

    companion object {
        @JvmStatic
        fun of(rental: Rental, paymentStatus: PaymentStatus?): RentalRejectResponse =
            RentalRejectResponse(rental.id, rental.status, rental.rejectReason, paymentStatus)
    }
}
